tests/cycpp_tests.py

```diff
@@ -1040,6 +1042,12 @@ def test_nuclide_uitype():
     f.given_classname = 'MyFactory'
     yield assert_raises, TypeError, f.impl
 
-
+def test_integration():
+    inf = os.path.join(os.path.dirname(__file__), 'cycpp_tests.h')
+    outf = tempfile.NamedTemporaryFile()
+    cmd = 'cycpp.py {} -o {}'.format(inf, outf.name)
+    p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=STDOUT, close_fds=True)
+    assert_equal('', p.stdout.read().decode())
+    
```

Comment:
The file is closed when garbage collected.

On Wed, Jun 3, 2015 at 12:38 PM, mbmcgarry <notifications@github.com> wrote:

> In tests/cycpp_tests.py
> <https://github.com/cyclus/cyclus/pull/1174#discussion_r31648592>:
>
> > @@ -1040,6 +1042,12 @@ def test_nuclide_uitype():
> >      f.given_classname = 'MyFactory'
> >      yield assert_raises, TypeError, f.impl
> >
> > -
> > +def test_integration():
> > +    inf = os.path.join(os.path.dirname(__file__), 'cycpp_tests.h')
> > +    outf = tempfile.NamedTemporaryFile()
> > +    cmd = 'cycpp.py {} -o {}'.format(inf, outf.name)
> > +    p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=STDOUT, close_fds=True)
> > +    assert_equal('', p.stdout.read().decode())
> > +
>
> Should the tempfile be closed/removed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1174/files#r31648592>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192